bdns/servers.go

```diff
@@ -99,10 +100,15 @@ func (sp *staticProvider) Stop() {}
 // addresses, and refreshes it regularly using a goroutine started by its
 // constructor.
 type dynamicProvider struct {
-	// The domain name which should be used for DNS. Will be used as the basis of
-	// a SRV query to locate DNS services on this domain, which will in turn be
-	// used as the basis for A queries to cache IP addrs for those services.
-	name string
+	// dnsAuthority is the single <hostname|IPv4|[IPv6]>:<port> of the DNS
+	// server to be used for resolution of DNS backends. If the address contains
+	// a hostname it will be resolved via the system DNS. If the port is left
+	// unspecified it will default to '53'.
+	dnsAuthority string
```

Comment:
It looks like this is allowed to be ""; we should document the behavior when that is the case.